subsys/ipc/ipc_service/backends/ipc_static_vrings.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2021 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ipc/ipc_static_vrings.h>
+#include <cache.h>
+
+#define SHM_DEVICE_NAME		"sram0.shm"
+
+#define VRING_ALIGNMENT		(4) /* Alignment of vring buffer */
+
+#define RPMSG_VQ_0		(0) /* TX virtqueue queue index */
+#define RPMSG_VQ_1		(1) /* RX virtqueue queue index */
+
+static void virtio_notify(struct virtqueue *vq)
+{
+	struct ipc_static_vrings *vr;
+
+	vr = CONTAINER_OF(vq->vq_dev, struct ipc_static_vrings, vdev);
+
+	if (vr->notify_cb) {
+		vr->notify_cb(vq, vr->priv);
+	}
+}
+
+#ifdef CONFIG_IPC_SERVICE_MASTER
+
+static void virtio_set_features(struct virtio_device *vdev, uint32_t features)
+{
+	/* No need for implementation */
+}
+
+static void virtio_set_status(struct virtio_device *p_vdev, unsigned char status)
+{
+	struct ipc_static_vrings *vr;
+
+	vr = CONTAINER_OF(p_vdev, struct ipc_static_vrings, vdev);
+
+	sys_write8(status, vr->status_reg_addr);
+	sys_cache_data_range((void *) vr->status_reg_addr,
+			     sizeof(status), K_CACHE_WB);
+}
+
+#endif /* #ifdef CONFIG_IPC_SERVICE_MASTER */
+
+static uint32_t virtio_get_features(struct virtio_device *vdev)
+{
+	return BIT(VIRTIO_RPMSG_F_NS);
+}
+
+static unsigned char virtio_get_status(struct virtio_device *p_vdev)
+{
+	struct ipc_static_vrings *vr;
+	uint8_t ret;
+
+	vr = CONTAINER_OF(p_vdev, struct ipc_static_vrings, vdev);
+
+	ret = VIRTIO_CONFIG_STATUS_DRIVER_OK;
+
+	if (!IS_ENABLED(CONFIG_IPC_SERVICE_MASTER)) {
```

Comment:
same remark, role to define per instance

Reply:
Yeah, I need to think how to implement this.